### PR TITLE
Adds tests to most functions

### PR DESCRIPTION
**Pull request by lucas-castillo**
Adds tests to most functions, as well as test coverage badge. Functions still without tests are ABS and some plotting functions we may want to get rid of. Relates to #5

### CODE REVIEW

File: src/mcrec.h [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+// Enable C++11 via this plugin (Rcpp 0.10.3 or later)
+// [[Rcpp::plugins("cpp11")]]
+
+#include "hmc_utils.h"
+#include "pdf_manage.h"
+
+Rcpp::List sampler_mc_rec_cpp(
+    Rcpp::NumericVector start,
+    int nChains, // for non-tempered versions of HMC and HOR, set this to one
```
Comment (yunxiao-li-psych) [inline src/mcrec.h:22]: Shall we change the name to `n_chains`?

File: src/test-cpp-samplers.cpp [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+      f,
+      false
+    );
+  }
+  
+  test_that("MCREC"){
+    Function f("rnorm");
+    List res = sampler_mc_rec_cpp(
+      NumericVector::create(0),// Rcpp::NumericVector start,
+      2, // int nChains,
```
Comment (yunxiao-li-psych) [inline src/test-cpp-samplers.cpp:108]: Same here (even though it is in the comments, we might want to keep it consistent.)

File: src/test-cpp-samplers.cpp [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    Function f("rnorm");
+    NumericVector one = {1};
+    NumericMatrix one_matrix(1, 1, one.begin());
+    NumericVector start_v = {1,1};
+    NumericMatrix start(2, 1, start_v.begin());
+    
+    // swap one
+    List res = sampler_mc3_cpp(
+      start, // Rcpp::NumericMatrix start,
+      2, // int nChains,
```
Comment (yunxiao-li-psych) [inline src/test-cpp-samplers.cpp:53]: Same here.

File: src/test-cpp-samplers.cpp [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+      NumericVector::create(1), //Rcpp::NumericVector weights,
+      f, //Rcpp::Function custom_func,
+      false, //bool useCustom,
+      .5 //double alpha=0
+    );
+    
+    // swap all
+    res = sampler_mc3_cpp(
+      start,// Rcpp::NumericMatrix start,
+      2, // int nChains,
```
Comment (yunxiao-li-psych) [inline src/test-cpp-samplers.cpp:71]: Same here.

File: tests/testthat/test-samplers.R [hunk 5]
```diff
@@ -1,3 +1,211 @@
...
+    sampler_mh(0, sigma_prop =  diag(1), custom_density = \(x){dnorm(x)})
+  )
+})
+
+test_that("mc3", {
+  expect_no_error(
+    sampler_mc3(start = matrix(c(0,0),ncol=1), distr_name = "norm", distr_params = c(0,1), sigma_prop = diag(1), nChains = 2)
+  )
+  expect_no_error(
+    sampler_mc3(start = matrix(c(0,0),ncol=1), custom_density = \(x){dnorm(x)}, sigma_prop = diag(1), nChains = 2)
```
Comment (yunxiao-li-psych) [inline tests/testthat/test-samplers.R:145]: Same here.

File: tests/testthat/test-samplers.R [hunk 6]
```diff
@@ -1,3 +1,211 @@
...
+  )
+  expect_no_error(
+    sampler_mc3(start = 0, custom_density = \(x){dnorm(x)}, sigma_prop = diag(1), nChains = 2)
```
Comment (yunxiao-li-psych) [inline tests/testthat/test-samplers.R:148]: Same here.

File: tests/testthat/test-samplers.R [hunk 7]
```diff
@@ -1,3 +1,211 @@
...
+  )
+  expect_error(
+    sampler_mc3(start = 0, distr_name = "norm", distr_params = c(0,1), sigma_prop = diag(1), nChains = 2.5)
```
Comment (yunxiao-li-psych) [inline tests/testthat/test-samplers.R:151]: Same here.

File: tests/testthat/test-samplers.R [hunk 8]
```diff
@@ -1,3 +1,211 @@
...
+  )
+  expect_error(
+    sampler_mc3(start = matrix(c(0,0,0),ncol=1), distr_name = "norm", distr_params = c(0,1), sigma_prop = diag(1), nChains = 2)
```
Comment (yunxiao-li-psych) [inline tests/testthat/test-samplers.R:154]: Same here.

File: tests/testthat/test-samplers.R [hunk 9]
```diff
@@ -1,3 +1,211 @@
...
+
+test_that("mchmc", {
+  expect_no_error(
+    sampler_mchmc(0, "norm", c(0,1))
+  )
+  expect_no_error(
+    sampler_mchmc(0, custom_density = \(x){dnorm(x)})
+  )
+  expect_error(
+    sampler_mchmc(0, "norm", c(0,1), nChains = 2.5)
```
Comment (yunxiao-li-psych) [inline tests/testthat/test-samplers.R:191]: Same here.

File: tests/testthat/test-samplers.R [hunk 10]
```diff
@@ -1,3 +1,211 @@
...
+
+test_that("sampler_mcrec", {
+  expect_no_error(
+    sampler_mcrec(0, "norm", c(0,1))
+  )
+  expect_no_error(
+    sampler_mcrec(0, custom_density = \(x){dnorm(x)})
+  )
+  expect_error(
+    sampler_mcrec(0, "norm", c(0,1), nChains = 2.5)
```
Comment (yunxiao-li-psych) [inline tests/testthat/test-samplers.R:206]: Same here.

**Review by yunxiao-li-psych — COMMENTED** [review]
The only concern is the name of `nChains`.

**Comment by lucas-castillo** [comment]
I'll merge and do `nChains` separately -- renaming the parameter is not part of adding tests

**Pull request merged**